geemap/common.py

```diff
@@ -16167,3 +16167,160 @@ def is_on_aws():
+        )
+
+    allowed_map_types = [
+        "roadmap",
+        "satellite",
+        "terrain",
+        "hybrid",
+        "traffic",
+        "streetview",
+    ]
+
+    # Support map type as a string with or without 'google.', such as 'Google Roadmap', 'Google.Roadmap', or 'Roadmap'
+    if isinstance(map_type, str):
+        map_type = map_type.lower().replace("google.", "").replace("google", "").strip()
+
+        if map_type not in allowed_map_types:
+            raise ValueError(
+                "mapType must be one of 'roadmap', 'satellite', 'terrain', 'hybrid', 'traffic', 'streetview'"
+            )
+
+    tile_args = {}
+
+    # Define the parameters for each map type
+    for m_type in allowed_map_types:
+
+        mapType = m_type
+        layerTypes = None
+
+        if m_type == "hybrid":
+            mapType = "satellite"
+            layerTypes = ["layerRoadmap"]
+        elif m_type == "terrain":
+            layerTypes = ["layerRoadmap"]
+        elif m_type == "traffic":
+            mapType = "roadmap"
+            layerTypes = ["layerTraffic"]
+        elif m_type == "streetview":
+            mapType = "roadmap"
+            layerTypes = ["layerStreetview"]
+
+        tile_args[m_type] = {
+            "mapType": mapType,
+            "language": language,
+            "region": region,
+            "layerTypes": layerTypes,
+            **kwargs,
+        }
+
+        if tile_args[m_type].get("layerTypes") is None:
+            del tile_args[m_type]["layerTypes"]
+
+    gmap_providers = {}
+
+    for m_type in allowed_map_types:
+
+        # If map_type is provided, only generate the specified map
+        if map_type is not None and m_type != map_type:
+            continue
+
+        args = tile_args[m_type]
+        response = requests.post(
+            f"https://tile.googleapis.com/v1/createSession?key={api_key}",
+            headers={"Content-Type": "application/json"},
+            json=args,
+        )
+
+        if response.status_code == 200:
+            res = response.json()
+            gmap_provider = TileProvider(
+                {
+                    "url": f"https://tile.googleapis.com/v1/2dtiles/{{z}}/{{x}}/{{y}}?session={res['session']}&key={{accessToken}}",
+                    "attribution": f"© Google {m_type.capitalize()}",
+                    "accessToken": api_key,
+                    "name": f"Google.{m_type.capitalize()}",
+                    "ext": res["imageFormat"],
+                    "tileSize": res["tileWidth"],
+                }
+            )
+            gmap_providers[m_type] = gmap_provider
+        else:
+            display(response.text)
+            gmap_provider = None
+
+    if map_type is None:  # Return all map types
+        return gmap_providers
+    else:  # Return the specified map type
+        return gmap_providers.get(map_type)
+
+
+def google_maps_api_key(token_name="MAPS_API_KEY"):
```

Comment:
Could you add type annotations to `google_maps_api_key`?

`lookup_key`/`token_name` should be `lookup_key: Optional[str] = "MAPS_API_KEY"`

Return type is `-> Optional[str]`